submissions/m-ruslan/memory-pair-game/app.js

```diff
@@ -1,51 +1,56 @@
-// main logic of game
-document.addEventListener("DOMContentLoaded", () => {
-  let cardID, cardValue, prevCardId, prevCardValue;
-  let qOfOpenCards = 0;
+let quantityqOfCardsLeft;
+let cardID, cardValue, prevCardId, prevCardValue;
+let quantityOfOpenCards = 0;
 
+document.addEventListener("DOMContentLoaded", () => {
   startNewGame();
 
   document.querySelector("main").addEventListener("click", (event) => {
     let aEl = event.target.closest(".flip-container");
-    if (!aEl) return;
-    if (aEl.id === prevCardId && aEl.id !== undefined) return;
+    if (!aEl || aEl?.id === prevCardId) return;
+    if (quantityOfOpenCards === 2) return;
 
-    if (qOfOpenCards < 2) {
-      qOfOpenCards++;
+    if (quantityOfOpenCards < 2) {
+      quantityOfOpenCards++;
       cardID = aEl.id;
       cardValue = aEl.getAttribute("value");
 
       flip(cardID);
+    }
+
+    if (quantityOfOpenCards === 1) {
+      prevCardId = cardID;
+      prevCardValue = cardValue;
+      return;
+    }
 
-      if (qOfOpenCards === 1) {
-        prevCardId = cardID;
-        prevCardValue = cardValue;
-      } else if (qOfOpenCards === 2) {
-        setTimeout(() => {
-          if (cardValue === prevCardValue && cardID !== prevCardId) {
-            match(cardID, prevCardId);
-          } else {
-            dontMatch(cardID, prevCardId);
-          }
-
-          qOfOpenCards = 0;
-          prevCardId = null;
-          prevValue = null;
-
-          if (qOfCardsLeft === 0) {
-            youWon();
-          }
-        }, 1000);
-      }
+    if (quantityOfOpenCards === 2) {
```

Comment:
wait i don't understand your logic. Can you explain why you return of cards equals 2, than you have same check but do another things

[if (quantityOfOpenCards === 2) return;](https://github.com/kottans/frontend-2021-homeworks/pull/348/files/4f72173d65b67338e2f855aa3a945533b90e2103..a8fea90d58784c9fb6a2ae31ba73ab79684fecfb#diff-51fc09a67585ef5dd14073d38db2ff93f05600af219b7161e849c62aed172c47R11)

Reply:
@babayK0 thank you for your question.
`if (quantityOfOpenCards === 2) return` in the beginning of the function is needed for situations when two card are already open, animation and timeout are still running, but user clicks on one more card. In this case we need this `return` to prevent from opening third card.
So the function is runned (after this `if (quantityOfOpenCards === 2) return`) only when 0 or 1 card was open before the current click,

Reply:
try to redo the logic for opening cards so that the code becomes obvious
because it's confusing.

Reply:
@babayK0 thank you. 
I've tried to redo the logic adding variable "moreCardsOpeningIsBlocked". Could you please check?)